csm_web/scheduler/views/section.py

```diff
@@ -779,3 +784,198 @@ def wotd(self, request, pk=None):
+            if section.mentor.user == request.user:
+                logger.error("Cannot `POST` swaps for a mentor.")
+                raise PermissionDenied("Cannot `POST` swaps for a mentor.")
+
+            '''Create a new Swap object between the the current student
+            and the student with the email specified in the request. This is essentially
+            creating a swap invite.'''
+            receiver_email = request.data.get("receiver_email", None)
+
+            # Check if receiver_email is empty
+            if receiver_email is None:
+                raise PermissionDenied("The `receiver_email` field in the request cannot be empty,"
+                                        "please specify a receiver.")
+
+            # Check if receiver_email is valid
+            try:
+                receiver = get_object_or_error(Student.objects, user__email=receiver_email)
+            except Exception as e:
+                raise NotFound("Invalid email provided.")
+
+            # Check if receiver is the sender
+            receiver_is_sender = receiver == student
+            if receiver_is_sender:
+                raise PermissionDenied("Cannot send a swap request to yourself.")
+
+            # Check if receiver has already sent a swap request to the sender
+            try:
+                incoming_swaps = Swap.objects.filter(receiver=student)
+            except Exception as e:
+                logger.error(e)
+                raise PermissionDenied("Could not fetch incoming swaps for student with id: " + 
+                                       str(student_id))
+            for swap in incoming_swaps:
+                if swap.sender == receiver:
+                    raise PermissionDenied("Cannot send a swap request to someone who has already"
+                                            "sent you a swap request.") 
+            # Create Swap request
+            try:
+                swap = Swap.objects.create(sender=student, receiver=receiver)
+            except Exception as e:
+                raise NotFound("Could not create swap for student with id: " + str(student_id))
+            # Successfuly created swap
+            logger.info(
+                "<Swap Created> User %s requested a swap with %s",
+                log_str(student.user),
+                log_str(receiver.user),
+            )
+            return Response({}, status=status.HTTP_201_CREATED)
+
+    
+    @action(detail=True, methods=["POST", "DELETE"], url_path='swap/(?P<swap_id>[^/.]+)')
+    def swap_with_id(self, request, section_id=None, swap_id=None):
+        '''If swap_id is not None, initiate the swap between the two students. This should 
+        be done in one atomic transaction. If either or both of the swaps are already 
+        deleted, then return a 404 error.'''
+        student_id = request.data.get("student_id", -1) if request.method != "GET" \
+            else request.query_params.get("student_id", -1)
```

Comment:
Since this inline conditional is quite long (it spans two lines), it's better convention (and more readable) to split it up into if/else blocks.

This should conventionally default to None if the field is not present, not -1. Additionally, the allowed request methods are only POST and DELETE, so the conditional doesn't quite make sense to me; the `else` part is never executed.